### PR TITLE
[SPARK-36421][SQL][DOCS] Use ConfigEntry.key to fix docs and set command results 

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes the issue that `ConfigEntry` to be introduced to the doc field directly without calling `.key`, which causes malformed documents on the web site and in the result of `SET -v`

1. https://spark.apache.org/docs/3.1.2/configuration.html#static-sql-configuration - spark.sql.hive.metastore.jars

2. set -v
image

### Why are the changes needed?

bugfix

### Does this PR introduce _any_ user-facing change?

no, but contains doc fix
### How was this patch tested?

new tests

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @gengliangwang

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveSQLConfSuite.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+
+class HiveSQLConfSuite extends TestHiveSingleton {
+
+  test("SPARK-36421: Validate all SQL configs to prevent from wrong use for ConfigEntry") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/HiveSQLConfSuite.scala:24]: this probably doesn't worth a new suite. how about just test it in `org.apache.spark.sql.hive.execution.SQLQuerySuite`?

**Review by cloud-fan — APPROVED** [review]

File: sql/hive/src/test/scala/org/apache/spark/sql/HiveSQLConfSuite.scala (hunk 1 above)
Comment (yaooqinn) [reply sql/hive/src/test/scala/org/apache/spark/sql/HiveSQLConfSuite.scala]: updated

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2.

**Pull request closed without merging**